### PR TITLE
ARROW-4288: [R] Add instructions for Ubuntu

### PR DESCRIPTION
**Pull request by krlmlr**
Need to use vendored Boost.

CC @romainfrancois.

### CODE REVIEW

**Comment by kou** [comment]
Could you open a JIRA issue? https://issues.apache.org/jira/projects/ARROW/issues

FYI: https://cwiki.apache.org/confluence/display/ARROW/Contributing+to+Apache+Arrow

File: r/README.Rmd [hunk 1]
```diff
@@ -24,8 +25,10 @@ First install a release build of the C++ bindings to arrow.
 git clone https://github.com/apache/arrow.git
 cd arrow/cpp && mkdir release && cd release
 
-# It is important to statically link to boost libraries
-cmake .. -DARROW_PARQUET=ON -DCMAKE_BUILD_TYPE=Release -DARROW_BOOST_USE_SHARED:BOOL=Off
+# It is important to statically link to boost libraries.
+# For this, you may need to use a vendored version of the
+# Boost libraries (e.g. on Ubuntu).
+cmake .. -DARROW_PARQUET=ON -DCMAKE_BUILD_TYPE=Release -DARROW_BOOST_USE_SHARED:BOOL=Off -DARROW_BOOST_VENDORED=ON
```
Comment (xhochy) [inline r/README.Rmd]: Vendoring Boost makes it risky to symbol collisions, we should try and namespace our own boost: https://issues.apache.org/jira/browse/ARROW-4286

**Comment by krlmlr** [comment]
Done. Note that this setup works for me (I can load the package and render the `README.Rmd` file), I'm sure there are better options.

**Comment by wesm** [comment]
I'm not sure about this as a default recommendation

* If you want to use Boost from the system package manager, then you should use `-DARROW_BOOST_USE_SHARED=ON`. I'm not sure why static linking was recommended here in the first place

* If you want to use statically-linked Boost, then either use the vendored Boost build, or obtain Boost from a different package manager (e.g. we use conda for a lot of other builds)

**Comment by wesm** [comment]
The default options now should be more compatible out of the box on Ubuntu. Closing this

https://github.com/apache/arrow/blob/master/cpp/cmake_modules/DefineOptions.cmake#L201

**Pull request closed without merging**